machinist/polling/metrics.go

```diff
@@ -43,14 +41,6 @@ func SendMetricsRequest(ctx context.Context, c *config.Node) error {
 		}, func() float64 {
 			return numErr
 		})
-		averageErr := atomic.NewFloat64(0.0)
-		promauto.NewGaugeFunc(prometheus.GaugeOpts{
-			Name:      "dmesg_errors_per_second",
```

Comment:
This is kinda goofy - usually one exports a counter, and does the rate via prometheus queries.

How about replacing this with a simple counter, and incrementing the counter inline below? Prometheus counters should be threadsafe with no other locking/atomics. Would that resolve the performance issue while preserving the metric?

Reply:
You're right is goofy, I made it look like the other ones. Still going to nix the average err metric because prometheus should be calculating that.